src/Controller.ts

```diff
@@ -69,9 +70,29 @@ export const Controller = {
 		const engines: RuleEngine[] = allEngines.filter(e => e.isEnabled());
 
 		if (engines.length == 0) {
-			throw SfdxError.create('@salesforce/sfdx-scanner', 'Controller', 'NoEnabledEnginesFound', [enginesToString(allEngines)]);
+			throw SfdxError.create('@salesforce/sfdx-scanner', 'Controller', 'NoEnabledEnginesFound', [servicesToString(allEngines)]);
 		}
 
 		return engines;
+	},
+
+	getAllDepCheckers: async(): Promise<DependencyChecker[]> => {
+		const depCheckers: DependencyChecker[] = [];
+		for (const depcheck of container.resolveAll<DependencyChecker>(Services.DependencyChecker)) {
+			await depcheck.init();
+			depCheckers.push(depcheck);
+		}
+		return depCheckers;
+	},
+
+	getEnabledDepCheckers: async (): Promise<DependencyChecker[]> => {
+		const allDepCheckers: DependencyChecker[] = await Controller.getAllDepCheckers();
+		const enabledCheckers: DependencyChecker[] = allDepCheckers.filter(d => d.isEnabled());
```

Comment:
isEnabled could be moved to the proposed `Service` interface, this code might be able to be refactored to find enabled services.

Reply:
Good point, I'll see if I can do that.